inc/html.php

```diff
@@ -1006,181 +385,86 @@ function html_li_default($item){
-                $ret .= "\n<ul class=\"$class\">\n";
+            for ($i = 0; $i < ($item['level'] - $level); $i++) {
+                if ($i) $html .= '<li class="clear">';
+                $html .= "\n".'<ul class="'.$class.'">'."\n";
                 $open++;
             }
             $level = $item['level'];
 
-        }elseif( $item['level'] < $level ){
+        } elseif ($item['level'] < $level) {
             //close last item
-            $ret .= "</li>\n";
-            while( $level > $item['level'] && $open > 0 ){
+            $html .= '</li>'."\n";
+            while ($level > $item['level'] && $open > 0 ) {
                 //close higher lists
-                $ret .= "</ul>\n</li>\n";
+                $html .= '</ul>'."\n".'</li>'."\n";
                 $level--;
                 $open--;
             }
-        } elseif ($ret !== '') {
+        } elseif ($html !== '') {
             //close previous item
-            $ret .= "</li>\n";
+            $html .= '</li>'."\n";
         }
 
         //print item
-        $ret .= call_user_func($lifunc,$item);
-        $ret .= '<div class="li">';
+        $html .= call_user_func($lifunc, $item);
+        $html .= '<div class="li">';
 
-        $ret .= call_user_func($func,$item);
-        $ret .= '</div>';
+        $html .= call_user_func($func, $item);
+        $html .= '</div>';
     }
 
     //close remaining items and lists
-    $ret .= "</li>\n";
-    while($open-- > 0) {
-        $ret .= "</ul></li>\n";
+    $html .= '</li>'."\n";
+    while ($open-- > 0) {
+        $html .= '</ul></li>'."\n";
     }
 
     if ($forcewrapper || $start_level < 2) {
         // Trigger building a wrapper ul if the first level is
         // 0 (we have a root object) or 1 (just the root content)
-        $ret = "\n<ul class=\"$class\">\n".$ret."</ul>\n";
+        $html = "\n".'<ul class="'.$class.'">'."\n".$html.'</ul>'."\n";
     }
 
-    return $ret;
+    return $html;
 }
 
 /**
  * display backlinks
  *
  * @author Andreas Gohr <andi@splitbrain.org>
  * @author Michael Klier <chi@chimeric.de>
+ * @deprecated 2020-07-18
  */
-function html_backlinks(){
-    global $ID;
-    global $lang;
-
-    print p_locale_xhtml('backlinks');
-
-    $data = ft_backlinks($ID);
-
-    if(!empty($data)) {
-        print '<ul class="idx">';
-        foreach($data as $blink){
-            print '<li><div class="li">';
-            print html_wikilink(':'.$blink,useHeading('navigation')?null:$blink);
-            print '</div></li>';
-        }
-        print '</ul>';
-    } else {
-        print '<div class="level1"><p>' . $lang['nothingfound'] . '</p></div>';
-    }
-}
-
-/**
- * Get header of diff HTML
- *
- * @param string $l_rev   Left revisions
- * @param string $r_rev   Right revision
- * @param string $id      Page id, if null $ID is used
- * @param bool   $media   If it is for media files
- * @param bool   $inline  Return the header on a single line
- * @return string[] HTML snippets for diff header
- */
```

Comment:
I assume `html_diff_head()` is deprecated as well?
`@deprecated 2020-07-18`

Reply:
re-declared once deleted functions in inc/html.php file in 87f229e. `html_diff_head()` now contains dbg_deprecated().